appservice/src/deploy/runWithZipStream.ts

```diff
@@ -43,15 +43,9 @@ export async function runWithZipStream(context: IActionContext, options: {
         ext.outputChannel.appendLog(vscode.l10n.t('Creating zip package...'), { resourceName: site.fullName });
         const zipFile: yazl.ZipFile = new yazl.ZipFile();
         let filesToZip: string[] = [];
-        let sizeOfZipFile: number = 0;
 
-        zipFile.outputStream.on('data', (chunk) => {
```

Comment:
I believe this 'data' event listener is the root cause of the deployment issue. During zip deployment we pass `zipFile.outputStream` to the request body and later the SDK calls pipe() on it to read the data and write it to the request body. Before that the SDK needs to acquire the access token. Once acquire token become async, there is chance that this event listener is called before pipe has been called towards the request body. When that happens, the data consumed by this event listener will be removed from the stream and the request body will never be able to receive them.

`yazl` internally maintains a cursor which gives the number of bytes it has written to the output stream. We can use that to report the number of bytes instead of using our own 'data' event listener.